app/src/pages/inside/testCaseLibraryPage/testCaseListWrapper/testCaseListWrapper.tsx

```diff
@@ -0,0 +1,170 @@
+  testCases: TestCase[];
+  loading: boolean;
+  setTestCases: (testCases: TestCase[] | ((prev: TestCase[]) => TestCase[])) => void;
+}
+
+export const TestCaseListWrapper = ({
+  testCases,
+  loading,
+  setTestCases,
+}: TestCaseListWrapperProps) => {
+  const { formatMessage } = useIntl();
+  const [activePage, setActivePage] = useState<number>(DEFAULT_CURRENT_PAGE);
+  const [pageSize, setPageSize] = useState<number>(DEFAULT_ITEMS_PER_PAGE);
+  const [filteredTestCases, setFilteredTestCases] = useState<TestCase[]>([]);
+  const [searchValue, setSearchValue] = useState<string>('');
+
+  const updateFilteredTestCases = useCallback(
+    (searchTerm: string = searchValue) => {
+      let filtered = testCases;
+
+      if (searchTerm.trim()) {
+        const term = searchTerm.toLowerCase();
+        filtered = filtered.filter(
+          (testCase) =>
+            testCase.name.toLowerCase().includes(term) ||
+            testCase.tags.some((tag) => tag.toLowerCase().includes(term)),
+        );
+      }
+
+      setFilteredTestCases(filtered);
+      setActivePage(DEFAULT_CURRENT_PAGE);
+    },
+    [testCases, searchValue],
+  );
+
+  useMemo(() => {
+    updateFilteredTestCases();
+  }, [updateFilteredTestCases]);
+
+  // Calculate pagination values
+  const totalItems = filteredTestCases.length;
+  const totalPages = Math.ceil(totalItems / pageSize);
+
+  const handleEdit = useCallback(() => {
+    // Here you would implement edit logic
+  }, []);
+
+  const handleDelete = useCallback(
+    (testCase: TestCase) => {
+      // Here you would typically show confirmation modal
+      const confirmationMessage = formatMessage(messages.deleteConfirmation, {
+        name: testCase.name,
+      });
+      if (window.confirm(confirmationMessage)) {
+        setTestCases((prev) => prev.filter((tc) => tc.id !== testCase.id));
+        setFilteredTestCases((prev) => prev.filter((tc) => tc.id !== testCase.id));
+      }
+    },
+    [formatMessage, setTestCases],
+  );
+
+  const handleDuplicate = useCallback(
+    (testCase: TestCase) => {
+      // Here you would implement duplication logic
+      const duplicatedTestCase: TestCase = {
+        ...testCase,
+        id: Date.now(),
+        name: `${testCase.name} ${formatMessage(messages.copySuffix)}`,
+      };
+      setTestCases((prev) => [...prev, duplicatedTestCase]);
+      setFilteredTestCases((prev) => [...prev, duplicatedTestCase]);
+    },
+    [formatMessage, setTestCases],
+  );
+
+  const handleMove = useCallback(() => {
+    // Here you would implement move logic (show modal with folder selection, etc.)
+  }, []);
+
+  const handleSearchChange = useCallback(
+    (targetSearchValue: string) => {
+      setSearchValue(targetSearchValue);
+      updateFilteredTestCases(targetSearchValue);
+    },
+    [updateFilteredTestCases],
+  );
+
+  const changePage = useCallback((page: number) => {
+    setActivePage(page);
+  }, []);
+
+  const changePageSize = useCallback((size: number) => {
```

Comment:
Nor `Pagination` nor its descendants are memoized. This useCallback has no sense and only increases memory used